clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -580,6 +582,41 @@ public void testConnectDisconnectDuringInSinglePoll() throws Exception {
         control.verify();
     }
 
+    @Test
+    public void testOutboundConnectionsCountInConnectionCreationMetric() throws Exception {
+        // create connections
+        int conns = 5;
+        InetSocketAddress addr = new InetSocketAddress("localhost", server.port);
+        for (int i = 0; i < conns; i++)
+            connect(Integer.toString(i), addr);
+
+        selector.poll(0L);
```

Comment:
We cannot guarantee that this poll will see all completed connections, so it would be better to poll in a loop until the total connections returned from`selector.connected()` after the poll equals `conns`.